src/IROperator.h

```diff
@@ -797,6 +797,26 @@ inline Tuple tuple_select(const Expr &c0, const Tuple &v0, const Expr &c1, const
 }
 // @}
 
+/** Oftentimes we want to pack a list of expressions with the same type
+ * into a channel dimension, e.g.,
+ * img(x, y, c) = select(c == 0, 100, // Red
+ *                       c == 1, 50,  // Green
+ *                               25); // Blue
+ * This is tedious when the list is long. The following function
+ * provide convinent syntax that allow one to write:
+ * img(x, y, c) = select_by_index(c, {100, 50, 25});
+ */
+// @{
+inline Expr select_by_index(const Expr &id, const std::vector<Expr> &values) {
+    user_assert(values.size() >= 2) << "select_by_index only accepts values with size >= 2.\n";
```

Comment:
It might be good to catch type mismatchs among the values early here, so the error is from `select_by_index`, rather than from the internal `select`.

Reply:
Done. Thanks!